opentelemetry-instrumentation/src/opentelemetry/instrumentation/_blobupload/api/blob.py

```diff
@@ -0,0 +1,125 @@
+            if isinstance(labels, dict):
+                self._labels.update(labels)
+            else:
+                for k in labels:
+                    self._labels[k] = labels[k]
+
+    @staticmethod
+    def from_data_uri(uri: str, labels: Optional[dict] = None) -> "Blob":
+        """Instantiate a blob from a 'data:...' URI.
+
+        Args:
+            uri: A URI in the 'data:' format. Supports a subset of 'data:' URIs
+              that encode the data with the 'base64' extension and that include
+              a content type. Should work with any normal 'image/jpeg', 'image/png',
+              'application/pdf', 'audio/aac', and many others. DOES NOT SUPPORT
+              encoding data as percent-encoded text (no "base64").
+
+            labels: Additional key/value data to include in the constructed Blob.
+        """
+        if not uri.startswith("data:"):
+            raise ValueError(
+                'Invalid "uri"; expected "data:" prefix. Found: "{}"'.format(
+                    uri
+                )
+            )
+        if ";base64," not in uri:
+            raise ValueError(
+                'Invalid "uri"; expected ";base64," section. Found: "{}"'.format(
+                    uri
+                )
+            )
+        data_prefix_len = len("data:")
+        after_data_prefix = uri[data_prefix_len:]
+        if ";" not in after_data_prefix:
+            raise ValueError(
+                'Invalid "uri"; expected ";" in URI. Found: "{}"'.format(uri)
+            )
+        content_type, remaining = after_data_prefix.split(";", 1)
+        while not remaining.startswith("base64,"):
+            _, remaining = remaining.split(";", 1)
+        assert remaining.startswith("base64,")
+        base64_len = len("base64,")
+        base64_encoded_content = remaining[base64_len:]
+        raw_bytes = base64.b64decode(base64_encoded_content)
+        return Blob(raw_bytes, content_type=content_type, labels=labels)
+
+    @property
+    def raw_bytes(self) -> bytes:
+        """Returns the raw bytes (payload) of this Blob."""
+        return self._raw_bytes
+
+    @property
+    def content_type(self) -> Optional[str]:
+        """Returns the content type (or None) of this Blob."""
+        return self._content_type
+
+    @property
+    def labels(self) -> Mapping[str, str]:
+        """Returns the key/value metadata of this Blob."""
+        return _frozendict(self._labels)
+
+    def __eq__(self, o):
+        return (
+            (isinstance(o, Blob)) and
+            (self.raw_bytes == o.raw_bytes) and
+            (self.content_type == o.content_type) and
+            (self.labels == o.labels)
+        )
+
+    def __repr__(self):
+        params = [repr(self._raw_bytes)]
+        if self._content_type is not None:
+            params.append("content_type={}".format(repr(self._content_type)))
+        if self._labels:
+            params.append("labels={}".format(json.dumps(self._labels, sort_keys=True)))
```

Comment:
why are we JSON formatting? this will be much slower, and not lead to the correct repr.